mimic/rest/fastly_api.py

```diff
@@ -11,6 +11,18 @@
 from mimic.canned_responses import fastly
 
 
+def text_urldata(urldata):
+    """
+    Convert ``urldata`` into text.
+
+    :param urldata: a dictionary mapping bytes to lists of bytes.
+
+    :return: a dictionary mapping text to lists of text.
+    """
+    return dict([(k.decode("utf-8"), [vv.decode("utf-8") for vv in v])
+                 for [k, v] in urldata.items()])
```

Comment:
I didn't want to get into refactoring this stuff too much because really we ought to be using the shiny new `URL` object available in Twisted 15.5+ for this sort of thing rather than manually dealing with URL encoding ourselves; you're right that it's repeated though.